### PR TITLE
ShouldFire-ify some inventory events

### PR DESCRIPTION
**Pull request by Aaron1011**
This PR checks some ShouldFire flags in some inventory packet listeners. This should greatly improve performance in the case where there are little to no plugins listening for inventory events.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/event/ShouldFire.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
     // You must always check a flag that either corresponds directly
     // to the event you're firing, or to a supertype of the event.
     // For example, when firing DropItemEvent.Dispense, you can check
-    // ShouldFire.DROP_ITEM_EVENT_DISPENSE or ShouldFire.SPAWN_ENTITY_EVENT
+    // ShouldFire.DROP_ITEM_EVENT_DISPENSE or ShouldFire    .SPAWN_ENTITY_EVENT
```
Comment (XakepSDK) [inline src/main/java/org/spongepowered/common/event/ShouldFire.java:44]: Extra spaces

**Review by gabizou — APPROVED** [review]
Pending verification from @Faithcaio this looks good.

**Comment by Faithcaio** [comment]
Code looks fine (have not tested)

**Review by ImMorpheus — APPROVED** [review]
Working as intended

**Pull request merged**